### PR TITLE
Update Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet.md

### PR DESCRIPTION
**Pull request by jonathan-jaffe-lmnd**
Grammatical corrections and improved readability.

Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [ ] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [ ] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries/actions?query=workflow%3A%22Markdown+Link+Check%22).
- [ ] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries/blob/master/CONTRIBUTING.md#markdown).
- [ ] All your assets are stored in the **assets** folder.
- [ ] All the images used are in the **PNG** format.
- [ ] Any references to websites have been formatted as [TEXT](URL)
- [ ] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [ ] The CI build of your PR pass, see the build status [here](https://github.com/OWASP/CheatSheetSeries/actions).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #<insert number here>.

Thank you again for your contribution :smiley:

### CODE REVIEW

File: cheatsheets/Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet.md [hunk 1]
```diff
@@ -45,18 +45,18 @@ But, suppose a user has a list of accounts, like a bank where database ID 23456
...
-Using a hash allow the following properties:
+Using a hash enables the following properties:
 
-- Do not require to maintain a mapping table (real ID vs front end ID) in user session or application level cache.
-- Makes creation of a collection a enumeration values more difficult to achieve because, even if attacker can guess the hash algorithm from the ID size, it cannot reproduce value due to the salt that is not tied to the hidden value.
+- Allows the absence of a mapping table—-real ID vs front-end ID—-in user session or application level caches.
+- Makes creation of the collection of enumeration values more difficult to achieve because, even if an attacker guesses the hash algorithm from the ID size, he cannot reproduce value, thanks to the salt.
 
-This is the implementation of the utility class that generate the identifier to use for exchange with the front end side:
+This is the implementation of the utility class that generates the the identifier to use for an exchange with the front-end:
```
Comment (szh) [inline cheatsheets/Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet.md:59]:
```suggestion
This is an implementation of the utility class that generates the identifier to use for an exchange with the front-end:
```

File: cheatsheets/Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet.md [hunk 2]
```diff
@@ -45,18 +45,18 @@ But, suppose a user has a list of accounts, like a bank where database ID 23456
...
 
-The proposal use a hash to replace the direct identifier. This hash is salted with a value defined at application level in order support topology in which the application is deployed in multi-instances mode (case for production).
+The proposal uses a hash to replace the direct identifier. This hash is salted with a value defined at the application level to support topologies in which the application is deployed in multi-instance mode, such as in production environments.
 
-Using a hash allow the following properties:
+Using a hash enables the following properties:
 
-- Do not require to maintain a mapping table (real ID vs front end ID) in user session or application level cache.
-- Makes creation of a collection a enumeration values more difficult to achieve because, even if attacker can guess the hash algorithm from the ID size, it cannot reproduce value due to the salt that is not tied to the hidden value.
+- Allows the absence of a mapping table—-real ID vs front-end ID—-in user session or application level caches.
```
Comment (szh) [inline cheatsheets/Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet.md:56]:
```suggestion
- Allows the absence of a mapping table (real ID vs front-end ID) in user session or application level caches.
```
I think parenthesis is a bit clearer.

File: cheatsheets/Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet.md [hunk 3]
```diff
@@ -2,19 +2,19 @@
 
 ## Introduction
 
-**I**nsecure **D**irect **O**bject **R**eference (called **IDOR** from here) occurs when a application exposes a reference to an internal implementation object. Using this way, it reveals the real identifier and format/pattern used of the element in the storage backend side. The most common example of it (although is not limited to this one) is a record identifier in a storage system (database, filesystem and so on).
+**I**nsecure **D**irect **O**bject **R**eference (called **IDOR** from here) occurs when a application exposes a reference to an internal implementation object. Using this method, an IDOR reveals the real identifier and format or pattern used of the element in the storage backend. The most common example is of a record identifier in a storage system such as a database or filesystem, though these are not the only examples.
```
Comment (szh) [inline cheatsheets/Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet.md:5]:
```suggestion
**I**nsecure **D**irect **O**bject **R**eference (called **IDOR** from here) occurs when a application exposes a reference to an internal implementation object. Using this method, an IDOR reveals the real identifier and format or pattern used of the element in the storage backend. The most common example is of a record identifier in a storage system such as a database or filesystem, though these are not the only examples.
```
There's a trailing space that's causing the lint action to fail.

**Review by szh — CHANGES_REQUESTED** [review]
Mostly good, just left a few nit comments.

**Comment by mackowski** [comment]
Hey @jonathan-jaffe-lmnd can you fix lint errors?

**Review by jmanico — APPROVED** [review]

**Pull request merged**